src/OpenLoco/CMakeLists.txt

```diff
@@ -0,0 +1,651 @@
+    set(MACOSX_BUNDLE_ICON_FILE "AppIcon")
+
+    if(${OPENLOCO_BRANCH} EQUAL master)
+        set(MACOSX_BUNDLE_SHORT_VERSION_STRING "${OPENLOCO_VERSION_TAG}")
+    else()
+        set(MACOSX_BUNDLE_SHORT_VERSION_STRING "${OPENLOCO_VERSION_TAG} ${OPENLOCO_BRANCH}")
+    endif()
+
+    set(BUNDLE_RESOURCES "")
+
+    find_program(ICONUTIL iconutil)
+    if (ICONUTIL)
+        file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/distribution/macos/AppIcon.iconset DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+        set(ICON_TARGET "${CMAKE_CURRENT_BINARY_DIR}/AppIcon.iconset")
+        set(ICON_OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/AppIcon.icns")
+        set(SOURCE_ICON_DIR "${CMAKE_CURRENT_SOURCE_DIR}/resources/logo")
+        add_custom_command(OUTPUT ${ICON_OUTPUT}
+                COMMAND cp icon_x16.png ${ICON_TARGET}/icon_16x16.png
+                COMMAND cp icon_x32.png ${ICON_TARGET}/icon_16x16@2x.png
+                COMMAND cp icon_x32.png ${ICON_TARGET}/icon_32x32.png
+                COMMAND cp icon_x64.png ${ICON_TARGET}/icon_32x32@2x.png
+                COMMAND cp icon_x128.png ${ICON_TARGET}/icon_128x128.png
+                COMMAND cp icon_x256.png ${ICON_TARGET}/icon_128x128@2x.png
+                COMMAND cp icon_x256.png ${ICON_TARGET}/icon_256x256.png
+                COMMAND cp icon_x512.png ${ICON_TARGET}/icon_256x256@2x.png
+                COMMAND cp icon_x512.png ${ICON_TARGET}/icon_512x512.png
+                COMMAND cp icon_x1024.png ${ICON_TARGET}/icon_512x512@2x.png
+                COMMAND ${ICONUTIL} --convert icns --output ${ICON_OUTPUT} ${ICON_TARGET}
+                WORKING_DIRECTORY ${SOURCE_ICON_DIR})
+        list(APPEND BUNDLE_RESOURCES ${ICON_OUTPUT})
+    endif ()
+
+    list(APPEND BUNDLE_RESOURCES ${CMAKE_CURRENT_SOURCE_DIR}/CHANGELOG.md)
+    list(APPEND BUNDLE_RESOURCES ${CMAKE_CURRENT_SOURCE_DIR}/CONTRIBUTORS.md)
+    set_target_properties(OpenLoco PROPERTIES RESOURCE "${BUNDLE_RESOURCES}")
+    target_sources(OpenLoco PUBLIC ${BUNDLE_RESOURCES})
+
+    file(GLOB BUNDLE_LANGUAGES "${CMAKE_CURRENT_SOURCE_DIR}/data/language/*")
+    target_sources(OpenLoco PUBLIC ${BUNDLE_LANGUAGES})
+    set_property(
+        SOURCE ${BUNDLE_LANGUAGES}
+        PROPERTY MACOSX_PACKAGE_LOCATION "Resources/language"
+    )
+endif ()
+
+# Add headers check to verify all headers carry their dependencies.
+# Only valid for Clang for now:
+# - GCC 8 does not support -Wno-pragma-once-outside-header
+# - Other compilers status unknown
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    add_library(OpenLoco-headers-check OBJECT ${OLOCO_HEADERS})
+    set_target_properties(OpenLoco-headers-check PROPERTIES LINKER_LANGUAGE CXX)
+    set_source_files_properties(${OLOCO_HEADERS} PROPERTIES LANGUAGE CXX)
+    loco_target_compile_link_flags(OpenLoco-headers-check)
+    target_compile_options(OpenLoco-headers-check PUBLIC -x c++ -Wno-pragma-once-outside-header -Wno-unused-const-variable)
+    target_link_libraries(OpenLoco-headers-check PUBLIC
```

Comment:
Did the previous version require linking as well?

Reply:
this is just to bring in the headers from the other targets. Yeah it previously needed this.